src/postgres/commands/connectPostgresDatabase.ts

```diff
@@ -21,9 +21,16 @@ export async function connectPostgresDatabase(context: IActionContext, treeItem?
     const oldTreeItemId: string | undefined = ext.connectedPostgresDB && ext.connectedPostgresDB.fullId;
     void ext.context.globalState.update(connectedPostgresKey, treeItem.fullId);
     ext.connectedPostgresDB = treeItem;
+    const username: string | undefined = (await treeItem.parent.getFullConnectionString()).username;
```

Comment:
I think that this `partialConnectionString` should already exist so you don't have to make any async calls to retrieve the username.

```suggestion
    const username: string | undefined = treeItem.parent.partialConnectionString.username;
```

Reply:
With the way that it is setup right now, when we load Postgres Servers the partialConnectionString is not automatically updated with the cached credentials. getFullConnectionString() is where credential values of partialConnectionString are updated.

Reply:
In that case, I recommend checking this property first before calling the async method.  We want to avoid any async calls unless they're absolutely necessary.